pkg/querier/blocks_store_queryable.go

```diff
@@ -392,7 +395,7 @@ func (q *blocksStoreQuerier) selectSorted(sp *storage.SelectHints, matchers ...*
 		}
 	}
 
-	return storage.NewMergeSeriesSet(seriesSets, storage.ChainedSeriesMerge), warnings, nil
+	return newTracingSet(q.ctx, "blocksStoreQuerier.seriesSet", storage.NewMergeSeriesSet(seriesSets, storage.ChainedSeriesMerge)), warnings, nil
```

Comment:
We should pass `spanCtx` instead of `q.ctx` here.

Reply:
I think `q.ctx` is correct, because `spanCtx` will finish when function returns. WDYT?

Reply:
Right, makes sense.